source/features/delete-review-comments-faster.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import {observe} from 'selector-observer';
+import {TrashIcon} from '@primer/octicons-react';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import onFragmentLoad from '../github-events/on-fragment-load';
+
+const deinit: VoidFunction[] = [];
+
+async function onButtonClick(event: delegate.Event): Promise<void> {
+	const comment = event.delegateTarget.closest<HTMLElement>('.js-comment')!;
+	await onFragmentLoad(select('include-fragment.SelectMenu-loading', comment), select('.timeline-comment-actions > details:last-of-type', comment)!);
+
+	select('.dropdown-menu .js-comment-delete > button', comment)!.click();
+}
+
+function addDeleteButton(cancelButton: Element): void {
+	cancelButton.classList.add('rgh-delete-button-added');
+	cancelButton.after(
+		<button className="btn btn-danger float-left rgh-review-comment-delete-button" type="button">
+			<TrashIcon/>
+		</button>
+	);
+}
+
+function init(): void {
+	const listener = delegate(document, '.rgh-review-comment-delete-button', 'click', onButtonClick);
+	const observer = observe('.review-comment > .unminimized-comment form:not(.js-single-suggested-change-form) .js-comment-cancel-button:not(.rgh-delete-button-added)', {
+		add: addDeleteButton
+	});
+	deinit.push(listener.destroy, observer.abort);
+}
+
+void features.add(__filebasename, {
+	include: [
+		pageDetect.isPRConversation,
+		pageDetect.isPRFiles
+	],
+	awaitDomReady: false,
+	init,
```

Comment:
Don't we need a `onetime` here?

Reply:
https://github.com/sindresorhus/refined-github/pull/4099#discussion_r614214617

Reply:
I still don't see why it should not need a one time...

On ajax the observe will be added again. Am I wrong?

Reply:
Because we have a `deinit` array https://github.com/sindresorhus/refined-github/pull/4072#issuecomment-792402458